config/backend-config/backend-config.go

```diff
@@ -164,6 +166,14 @@ func (bc *CommonBackendConfig) configUpdate(ctx context.Context, statConfigBacke
 }
 
 func (bc *CommonBackendConfig) pollConfigUpdate(ctx context.Context, workspaces string) {
+	bc.pollConfigUpdatefunclock.Lock()
```

Comment:
`pollConfigUpdate` should not be called twice in the first place, but it makes sense to make the system more robust either way.

I would suggest [sync.Once](https://pkg.go.dev/sync#Once) as a more elegant way to do this.

Reply:
Yeah. Was in a hurry. Will clean it up.